lib/clone.js

```diff
@@ -86,6 +87,16 @@ module.exports = internals.clone = function (obj, options = {}, _seen = null) {
             continue;
         }
 
+        // Can only be covered in node 21+
```

Comment:
`structuredClone` is available from v18.x. v21 is the one that breaks the stack property copying old behaviour.

Reply:
Does it matter if we use structuredClone early though? I guess it can't hurt, right?

Reply:
Functionally I would not expect it to, though it might change the performance characteristics. It could be significantly slower (eg. from changing the prototype). Hopefully something like https://github.com/hapijs/boom/pull/304 will be part of the new releases to avoid the copying altogether.

Reply:
This appears to break boom when using jest, because for jest `structuredClone(new Error('test')) instanceof Error` is false 🤦🏻‍♂️ Not a problem for everyone, but I expect some bug reports...

Reply:
> This appears to break boom when using jest, because for jest `structuredClone(new Error('test')) instanceof Error` is false 🤦🏻‍♂️ Not a problem for everyone, but I expect some bug reports...

That does not seem true in general. Especially because we override the prototype if it doesn't match the input. How are you using it?

Reply:
Doing a `Boom.badRequest(err)` fails when used in jest. Boom errors themselves are not a concern.

Reply:
Indeed, I had to switch to the latter on my project, it's not difficult but not something I expected to do either. Anyway, I think we can expect some complaints.

Reply:
I don't know if it's specific to my computer, but creating an entirely new project with only jest and boom with a super simple test already fails, yet I can't reproduce that in stackblitz for example 🤷🏻‍♂️ 
![image](https://github.com/user-attachments/assets/fa89f25d-3849-40e9-b55f-37a08887bcdc)

Reply:
I figured it out! We assume that the cloned class is the same as the original class, which for plain Errors under this kind of VM is not true. I have made a fix for master in #399.